search-api/src/main/scala/no/ndla/searchapi/service/search/IndexService.scala

```diff
@@ -132,9 +132,9 @@ trait IndexService[D <: Content](using
       indexingBundle: IndexingBundle
   ): Try[D] = {
     for {
-      _       <- createIndexIfNotExists()
-      request <- createIndexRequest(imported, searchIndex, indexingBundle)
-      _       <- e4sClient.execute(request)
+      _            <- createIndexIfNotExists()
+      maybeRequest <- createIndexRequest(imported, searchIndex, indexingBundle)
+      _ = maybeRequest.map(e4sClient.execute(_))
```

Comment:
Denne vil vel ikke feile nå dersom `execute` feiler?
Så vi må vel enten gjøre 


```suggestion
      _            <- maybeRequest match {
        case Some(req) => e4sClient.execute(req)
        case None      => Success(())
      }
```


eller kanskje lage `e4sClient.maybeExecute(...)`?

```scala
  def maybeExecute[T, U](
      maybeRequest: Option[T]
  )(implicit handler: Handler[T, U], ct: ClassTag[U]): Try[Option[RequestSuccess[U]]] =
    maybeRequest match {
      case Some(request) => execute(request).map(Some(_))
      case None          => Success(None)
    }
```